Update CI config

JStech opened the pull request:
* Use industrial_ci UNDERLAY again, after learning that we did #76 wrong.
* Unify config with the one from main MoveIt repo

rhaschke reviewed (commented):
I took the chance to unify the CI config with that of the MoveIt main repo.
I also added a Noetic build. However, this fails. Please have a look!

tylerjw commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,64 @@
+
+name: CI
+
+on: [push, pull_request]
```
While updating this, do we want to add the pre-release test here too?

tylerjw commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,64 @@
+          - IMAGE: melodic
+            CCOV: true
+            CATKIN_LINT: true
+            UPSTREAM_WORKSPACE: 'upstream.rosinstall'
```
I believe this is needed in noetic also because it depends on recent changes in `rviz_visual_tools`.  See https://github.com/ros-planning/moveit_calibration/blob/master/upstream.rosinstall

@JStech is this true or does the released noetic version of `rviz_visual_tools` work for this?

JStech replied:
Honestly I'm not sure. I'll see what I can figure out.

JStech replied:
Yes, this does build on noetic using the released `rviz_visual_tools`.

JStech commented:
> I also added a Noetic build. However, this fails. Please have a look!

That test failure is because OpenCV 3.2, the default version for Ubuntu 18.04, has a buggy ArUco board pose detector. See discussion here, #5. I added a check to use the incorrect pose for 3.2 and the correct pose otherwise.

rhaschke commented on moveit_calibration_plugins/handeye_calibration_target/test/handeye_target_aruco_test.cpp:
```diff
@@ -140,8 +140,14 @@ TEST_F(MoveItHandEyeTargetTester, DetectArucoMarkerPose)
   std::cout << "Translation:\n"
             << ret.translation() << "\nRotation:\n"
             << ret.rotation().eulerAngles(0, 1, 2) << std::endl;
+#if CV_VERSION_MAJOR == 3 && CV_VERSION_MINOR == 2
+  // OpenCV 3.2.0 (which is the default on Ubuntu 18.04) has a buggy ArUco pose detection implementation
   Eigen::Vector3d t(0.412949, -0.198895, 11.1761);
   Eigen::Vector3d r(0.324172, -2.03144, 2.90114);
+#else
+  Eigen::Vector3d t(0.0148984, 0.0123107, 0.58609);
+  Eigen::Vector3d r(2.12328, -1.50481, -1.29729);
+#endif
```
If OpenCV 3.2 is that much off, we should add a warning about that issue in the cmake file.

gavanderhoorn replied:
Yes, that would be *very* much appreciated.

Something like:

```cmake
message(WARNING "Your version of OpenCV (3.2) is known to have buggy/underperforming Aruco detection!")
```

JStech replied:
Thanks for the suggestion--I'll add the warning.

JStech replied:
I also put a stronger statement in the README about this and recommended ChArUco boards generally in #85--I've done some experiments and they're definitely better. I need to make a blog post about it.

rhaschke reviewed (approved):
From my point of view, this is ready to merge.
I'm not sure though, that we should consider wrong results for the unit tests against OpenCV 3.2. Alternatively, we could skip the test completely (in this case). However, then the remaining code isn't tested as well. Hence, I'm fine with the current solution as well.
